tune_sklearn/tune_basesearch.py

```diff
@@ -735,18 +739,37 @@ def _fill_config_hyperparam(self, config):
         """
         raise NotImplementedError("Define in child class")
 
-    def _tune_run(self, config, resources_per_trial):
+    def _tune_run(self, config, resources_per_trial, tune_params=None):
         """Wrapper to call ``tune.run``. Implement this in a child class.
 
         Args:
             config (:obj:`dict`): dictionary to be passed in as the
                 configuration for `tune.run`.
             resources_per_trial (:obj:`dict` of int): dictionary specifying the
                 number of cpu's and gpu's to use to train the model.
+            tune_params (dict, optional): User defined parameters passed to
+                ``tune.run``.
 
         """
         raise NotImplementedError("Define in child class")
 
+    def _override_run_args_with_tune_params(self, run_args, tune_params):
+        """Helper method to override tune.run run arguments with user
+        supplied dict"""
+
+        if tune_params:
+            user_overrides = {k for k in tune_params if k in run_args}
+            if user_overrides:
+                warnings.warn(
+                    "The following preset tune.run parameters will "
+                    f"be overriden by tune_params: {', '.join(user_overrides)}"
+                    ". This may cause unexpected issues! If you experience "
+                    "issues, please try removing those parameters from "
+                    "tune_params.")
+            warnings.warn("Using user supplied tune_params.")
```

Comment:
hmm, if a user passes this in, they wouldn't expect a warning here right?

Reply:
This probably should be an logging info message instead of a warning, now that I think about it